test/test-env-vars.c

```diff
@@ -86,5 +88,19 @@ TEST_IMPL(env_vars) {
   r = uv_os_unsetenv(name);
   ASSERT(r == 0);
 
+  /* Check getting all env variables. */
+  r = uv_os_environ(&envitems, &envcount);
+  ASSERT(r == 0);
+  ASSERT(envcount > 0);
+
+  for (i = 0; i < envcount; i++) {
+    /* printf("Env: %s = %s\n", envitems[i].name, envitems[i].value); */
+    if (strcmp(envitems[i].name, "PATH") == 0)
+      break;
+  }
+  ASSERT(i < envcount);
```

Comment:
Maybe it's a good idea setting a variable before calling `uv_os_environ` and check that variable is correctly retrieved. Also checking for an empty variable might be also good.

Reply:
👍